### ISSUE TITLE
Fetching 10000 docs in parallel is failing with seaweedfs storage

### ISSUE THREAD

**Issue by fmassot** [quickwit-oss/quickwit#3273]
A user reported the following error when executing 5 requests in parallel. Each request fetches 10 000 docs.

```bash

[2m2023-05-05T09:26:20.710Z[0m [31mERROR[0m [1mfetch_docs[0m[2m:[0m [2mquickwit_search::fetch_docs[0m[2m:[0m Error when fetching docs in splits. [3msplit_ids[0m[2m=[0m["01GYF19T9WRH371R8APMAPDA3R", "01GYF4D88AHBYQ9YATRRFS1XNF", "01GYFBSFKEKWYW7J6CE3H28HZT", "01GYF13J00NP6DDARY2PA8XS96", "01GYF2NB3XKQAH1X3M2NBZ4P2J", ....
 
Caused by:
    An IO error occurred: 'Error obtaining chunk: error reading a body from connection: end of file before message length reached Ctx:Failed to fetch slice 914729148..914974812 for object: s3://quickwit/indexes/index/01GYFADW547Q67QA5KPHPXYT5Q.split'

```

When fetching 1000 documents, there are no errors.

This is probably linked to the number of requests we are making on the storage. Currently, we limit the number of requests per split to `NUM_CONCURRENT_REQUESTS`. But in the case where there are many splits and many requests  in parallel, we have no limitation.

@trinity-1686a is already working on optimization fetch docs requests, it's worth waiting for the merge of @trinity-1686a 
 work and then tackling this issue.

**Comment by fulmicoton** [comment]
We can probably add some configurable limit on the number of concurrent fetch request. A global semaphore will do the trick.

**Comment by trinity-1686a** [comment]
fixed by #3356
